### ISSUE TITLE
Groups are broken for regexes with in-line flags since 0.6.21

### ISSUE THREAD

**Issue by Aurora12** [scala-js/scala-js#3406]
```
    val a = "(?i)(a)".r
    val b = a.findAllMatchIn("aA")
    println(s"TEST: ${b.map(_.matched).mkString(",")}")
```
This, compiled with 0.6.21-24 (Scala 2.12.4), crashes at runtime on [GroupStartMap.scala](https://github.com/scala-js/scala-js/blob/master/javalib/src/main/scala/java/util/regex/GroupStartMap.scala) line 405: **"[Internal error] Executed '/((?:(?:((?:((?:)?)((?:i))))(a))))/gi' on 'aA' at position 0, got an error. Original pattern '(?i)(a)' did match however."**

When I remove either flags or grouping, the code works as expected.
The same project compiled with 0.6.20 works fine.

**Comment by sjrd** [comment]
Thanks for the report.

**Comment by sjrd** [comment]
Fixed in d739fe6ec7d36e0f346381b36b675c25ec07bd69.
